aws/resource_aws_redshift_snapshot_schedule_association.go

```diff
@@ -0,0 +1,242 @@
+				if err != nil {
+					return nil, fmt.Errorf("Error parse Redshift Cluster Snapshot Schedule Association ID %s: %s", d.Id(), err)
+				}
+
+				d.Set("cluster_identifier", clusterIdentifier)
+				d.Set("schedule_identifier", scheduleIdentifier)
+				d.SetId(fmt.Sprintf("%s/%s", clusterIdentifier, scheduleIdentifier))
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster_identifier": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"schedule_identifier": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsRedshiftSnapshotScheduleAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).redshiftconn
+	clusterIdentifier := d.Get("cluster_identifier").(string)
+	scheduleIdentifier := d.Get("schedule_identifier").(string)
+
+	_, err := conn.ModifyClusterSnapshotSchedule(&redshift.ModifyClusterSnapshotScheduleInput{
+		ClusterIdentifier:    aws.String(clusterIdentifier),
+		ScheduleIdentifier:   aws.String(scheduleIdentifier),
+		DisassociateSchedule: aws.Bool(false),
+	})
+
+	if err != nil {
+		return fmt.Errorf("Error associating Redshift Cluster (%s) and Snapshot Schedule (%s): %s", clusterIdentifier, scheduleIdentifier, err)
+	}
+
+	if err := waitForRedshiftSnapshotScheduleAssociationActive(conn, 75*time.Minute, clusterIdentifier, scheduleIdentifier); err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", clusterIdentifier, scheduleIdentifier))
+
+	return resourceAwsRedshiftSnapshotScheduleAssociationRead(d, meta)
+}
+
+func resourceAwsRedshiftSnapshotScheduleAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).redshiftconn
+	clusterIdentifier, scheduleIdentifier, err := resourceAwsRedshiftSnapshotScheduleAssociationParseId(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parse Redshift Cluster Snapshot Schedule Association ID %s: %s", d.Id(), err)
+	}
+
+	descOpts := &redshift.DescribeSnapshotSchedulesInput{
+		ClusterIdentifier:  aws.String(clusterIdentifier),
+		ScheduleIdentifier: aws.String(scheduleIdentifier),
+	}
+
+	resp, err := conn.DescribeSnapshotSchedules(descOpts)
+	if err != nil {
+		return fmt.Errorf("Error describing Redshift Cluster %s Snapshot Schedule %s: %s", clusterIdentifier, clusterIdentifier, err)
+	}
+
+	if resp.SnapshotSchedules == nil || len(resp.SnapshotSchedules) == 0 {
+		return fmt.Errorf("Unable to find Redshift Cluster (%s) Snapshot Schedule (%s) Association", clusterIdentifier, scheduleIdentifier)
+	}
+	snapshotSchedule := resp.SnapshotSchedules[0]
+	if snapshotSchedule.AssociatedClusters == nil || aws.Int64Value(snapshotSchedule.AssociatedClusterCount) == 0 {
+		return fmt.Errorf("Unable to find Redshift Cluster (%s)", clusterIdentifier)
+	}
+
+	var associatedCluster *redshift.ClusterAssociatedToSchedule
+	foundAssociatedCluster := false
```

Comment:
Nit: This boolean variable is extraneous since we can check `associatedCluster == nil` 👍

Reply:
Thanks, fix it.